lib/codegen/Decision.cpp

```diff
@@ -380,6 +383,33 @@ llvm::Value *Decision::getTag(llvm::Value *val) {
   return res;
 }
 
+static void initChoiceBuffer(DecisionNode *dt, llvm::Module *module, llvm::BasicBlock *block, llvm::BasicBlock *stuck, llvm::BasicBlock *fail, llvm::AllocaInst **choiceBufferOut, llvm::AllocaInst **choiceDepthOut, llvm::IndirectBrInst **jumpOut) {
+  FailNode::get()->predecessors.clear();
+  FailNode::get()->successors.clear();
+  std::set<LeafNode *> leaves;
+  dt->preprocess(leaves);
+  dt->computeLiveness(leaves);
+  auto ty = llvm::ArrayType::get(llvm::Type::getInt8PtrTy(module->getContext()), dt->getChoiceDepth() + 1);
+  llvm::AllocaInst *choiceBuffer = new llvm::AllocaInst(ty, 0, "choiceBuffer", block);
+  llvm::AllocaInst *choiceDepth = new llvm::AllocaInst(llvm::Type::getInt64Ty(module->getContext()), 0, "choiceDepth", block);
+  new llvm::StoreInst(llvm::ConstantInt::get(llvm::Type::getInt64Ty(module->getContext()), 0), choiceDepth, block);
+  auto zero = llvm::ConstantInt::get(llvm::Type::getInt64Ty(module->getContext()), 0);
```

Comment:
nit: you could define the `zero` constant above the previous line and use it there too